src/site/content/en/identity/passkey-registration/index.md

```diff
@@ -16,166 +17,318 @@ tags:
+[Device Support - passkeys.dev](https://passkeys.dev/device-support/) to learn 
+what combination of browsers and an operating systems support creating a 
+passkey.
 
 ## Create a new passkey
 
+Here's how a frontend should operate upon a request to create a new passkey.
+
 ### Feature detection
 
-Before prompting the user to create a new passkey, check if:
+Before displaying a "Create a new passkey" button, check if:
 
--   The browser supports [WebAuthn](https://w3c.github.io/webauthn/).
--   The device supports a platform authenticator.
--   The browser supports conditional UI.
+* The browser supports WebAuthn.
+* The device supports a platform authenticator (can create a passkey and 
+  authenticate with the passkey).
+* The browser supports [WebAuthn conditional UI](/passkey-form-autofill/).
 
-```js
-// Availability of `window.PublicKeyCredential` means WebAuthn is usable.
-if (window.PublicKeyCredential &&
-    PublicKeyCredential.isUserVerifyingPlatformAuthenticatorAvailable &&
-    PublicKeyCredential.isConditionalMediationAvailable) {
-  // Check if user verifying platform authenticator is available.
-  Promise.all([
-    PublicKeyCredential.isUserVerifyingPlatformAuthenticatorAvailable(),
-    PublicKeyCredential.isConditionalMediationAvailable(),
-  ]).then(results => {
-    if (results.every(r => r === true)) {
-      // Call WebAuthn creation
-    }
-  });
-}
+```js  
+// Availability of `window.PublicKeyCredential` means WebAuthn is usable.  
+// `isUserVerifyingPlatformAuthenticatorAvailable` means the feature detection is usable.  
+// `​​isConditionalMediationAvailable` means the feature detection is usable.  
+if (window.PublicKeyCredential &&  
+    PublicKeyCredential.isUserVerifyingPlatformAuthenticatorAvailable &&  
+    PublicKeyCredential.​​isConditionalMediationAvailable) {  
+  // Check if user verifying platform authenticator is available.  
+  Promise.all([  
+    PublicKeyCredential.isUserVerifyingPlatformAuthenticatorAvailable(),  
+    PublicKeyCredential.​​isConditionalMediationAvailable(),  
+  ]).then(results =&gt; {  
+    if (results.every(r =&gt; r === true)) {  
+      // Display "Create a new passkey" button  
+    }  
+  });  
+}  
 ```
 
+Until all the conditions have been met, passkeys will not be supported on this browser. 
+The "Create a new passkey" button shouldn't be displayed until then.
+
+### Fetch important information from the backend
+
+When the user clicks the button, fetch important information to call 
+`navigator.credentials.create()` from the backend:
+
+* **[`challenge`](https://w3c.github.io/webauthn/#dom-publickeycredentialcreationoptions-challenge)**: 
+  A server-generated challenge in ArrayBuffer for this registration. This is 
+  required but unused during registration unless doing 
+  [attestation](https://developer.mozilla.org/docs/Web/API/Web_Authentication_API/Attestation_and_Assertion)—an 
+  advanced topic not covered here. Absent attestation, it can be set to `new 
```

Comment:
The "Absent attestation, it can be set to `newUint8Array([0])`, but never do this during sign-in." sentence is very confusing to me, I absolutely didn't understand what it means.

Reply:
I'd rather remove this part as this is optional and rather confusing.